tests/negative_tests.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+use std::os::unix::{io::AsRawFd, net::UnixStream};
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc,
+};
+
+use event_manager::{EventManager, EventOps, EventSubscriber, Events, SubscriberOps};
+use vmm_sys_util::epoll::EventSet;
+
+#[derive(Debug)]
+struct UnixStreamSubscriber {
+    stream: UnixStream,
+    rhup_count: AtomicU64,
```

Comment:
Is this an atomic just to have interior mutability and implement `EventSubscriber` instead of `MutEventSubscriber` for the object? If so, would it make sense to avoid the `Arc` in the `EventManager` signature below and use regular mutability?

Reply:
That is not the only reason for using `Arc`. I am using because it is easier to just clone the subscriber before moving its ownership in the`EventManager` so that I can later on retrieve the number of rhups.

Reply:
got it